crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1908,29 +1908,47 @@ impl Room {
             }
             self.send_state_event(content).await?;
 
+            // Spin on the sync beat event, since the first sync we receive might not
+            // include the encryption event.
+            //
             // TODO do we want to return an error here if we time out? This
             // could be quite useful if someone wants to enable encryption and
             // send a message right after it's enabled.
-            _ = timeout(self.client.inner.sync_beat.listen(), SYNC_WAIT_TIME).await;
+            let res = timeout(
+                async {
+                    loop {
+                        // Listen for sync events, then check if the encryption state has been set.
+                        self.client.inner.sync_beat.listen().await;
+                        let _sync_lock = self.client.base_client().sync_lock().lock().await;
+                        if self.inner.encryption_state().is_encrypted() {
```

Comment:
Are we not interested in **any** encryption state that's not `Unknown` here? 🤔

Reply:
You make a valid point! I shall change.